src/lib/election.ts

```diff
@@ -0,0 +1,95 @@
+import { atom, subscribe, transaction } from "atom.io"
+import { doc, onSnapshot } from "firebase/firestore"
+
+import type { ElectionData } from "../types"
+import { myselfSelector } from "./auth"
+import { db } from "./firebase"
+import { actualVoteSelectors, serializedVoteAtoms } from "./votes"
+
+export const currentElectionIdAtom = atom<string | null>({
+	key: `currentElectionId`,
+	default: null,
+})
+
+export const currentElectionLabelAtom = atom<string | null>({
+	key: `currentElectionLabel`,
+	default: null,
+})
+
+export const electionAtom = atom<ElectionData>({
+	key: `election`,
+	default: {
+		id: ``,
+		name: ``,
+		state: `not-started`,
+		createdAt: new Date(),
+		createdBy: ``,
+		users: [],
+		label: ``,
+		title: ``,
+		subtitle: ``,
+	},
+	effects: [
+		({ setSelf }) => {
+			let unSub: (() => void) | undefined
+			subscribe(currentElectionIdAtom, ({ newValue }) => {
+				unSub?.()
+				if (newValue === null) {
+					return
+				}
+				unSub = onSnapshot(doc(db, `elections`, newValue), (snapshot) => {
+					const election = snapshot.data() as ElectionData
+					setSelf(election)
+				})
+			})
+			return unSub
+		},
+	],
+})
+
+export const retractSubmittedBallotTX = transaction<() => void>({
+	key: `retractSubmittedBallot`,
+	do: ({ get, set }) => {
+		const myId = get(myselfSelector)?.id
+		if (!myId) {
+			console.error(`Attempted to retract a ballot but you are not logged in`)
+			return
+		}
+		const serializedVote = get(serializedVoteAtoms, myId)
+		if (serializedVote.finished) {
+			set(serializedVoteAtoms, myId, {
+				...serializedVote,
+				finished: false,
+			})
```

Comment:
Trying to understand transactions a little here: There is only a single `set()` in this function. Is it still necessary to make this a transaction?